application/src/main/java/org/opentripplanner/updater/trip/metrics/StreamingTripUpdateMetrics.java

```diff
@@ -79,6 +90,54 @@ private void setFailures(UpdateResult result) {
     }
   }
 
+  private void setFailuresByTypeAndProvider(UpdateResult result) {
+    Map<Pair<String, UpdateError.UpdateErrorType>, Long> failureCountByTypeAndProvider = result
+      .errors()
+      .stream()
+      .collect(
+        Collectors.groupingBy(
+          error -> Pair.of(error.provider(), error.errorType()),
+          Collectors.counting()
+        )
+      );
+
+    for (var entry : failureCountByTypeAndProvider.entrySet()) {
+      Counter counter = failuresByTypeAndProvider.get(entry.getKey());
+      if (Objects.isNull(counter)) {
+        counter =
+          getCounter(
+            "failures_by_type_and_provider",
+            "Total failed trip updates by type and provider",
+            Tag.of("provider", entry.getKey().getLeft()),
+            Tag.of("errorType", entry.getKey().getRight().name())
+          );
+        failuresByTypeAndProvider.put(entry.getKey(), counter);
+      }
+      counter.increment(entry.getValue());
```

Comment:
I'm the original author of this class and I think I made a mistake back then. Is it possible to add tags for each invocation of increment()?

Reply:
Okay, I've been doing it wrong back then and to have dynamic tags you should create a new one every time: https://stackoverflow.com/questions/59592118/dynamic-tag-values-for-the-counter-metric-in-micrometer

:facepalm: 

This is also what the built-in Jersey instrumentation does: https://github.com/micrometer-metrics/micrometer/blob/main/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jersey/server/MetricsRequestEventListener.java#L124-L141

I think we should chat in the dev meeting to figure out a better strategy.

Reply:
Sure, I'll put it on the agenda.

Looks to me like it's safe to instantiate them every time.